internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -62,7 +77,199 @@ func (r *ClusterDeploymentsReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	return ctrl.Result{}, nil
+	c, err := r.Installer.GetClusterByKubeKey(req.NamespacedName)
+
+	if gorm.IsRecordNotFoundError(err) {
+		return r.createNewCluster(ctx, req.NamespacedName, cluster)
+	}
+
+	return r.updateState(ctx, cluster, c, nil)
+}
+
+func (r *ClusterDeploymentsReconciler) createNewCluster(
+	ctx context.Context,
+	key types.NamespacedName,
+	cluster *hivev1.ClusterDeployment) (ctrl.Result, error) {
+
+	r.Log.Infof("Creating a new cluster %s %s", cluster.Name, cluster.Namespace)
+	spec := cluster.Spec
+
+	pullSecret, err := getPullSecret(ctx, r.Client, spec.PullSecretRef.Name, key.Namespace)
+	if err != nil {
+		r.Log.WithError(err).Error("failed to get pull secret")
+		return ctrl.Result{}, nil
+	}
+
+	clusterParams := &models.ClusterCreateParams{
+		//AdditionalNtpSource:      swag.String(spec.AdditionalNtpSource), // TODO: get from AgentEnvSpec
+		//HTTPProxy:                swag.String(spec.HTTPProxy), // TODO: get from AgentEnvSpec
+		//HTTPSProxy:               swag.String(spec.HTTPSProxy), // TODO: get from AgentEnvSpec
+		//NoProxy:                  swag.String(spec.NoProxy), // TODO: get from AgentEnvSpec
+		//SSHPublicKey:          spec.SSHPublicKey, // TODO: get from AgentEnvSpec
+		BaseDNSDomain:    spec.BaseDomain,
+		Name:             swag.String(spec.ClusterName),
+		OpenshiftVersion: swag.String("4.7"), // TODO: check how to set openshift version
+		Operators:        nil,                // TODO: handle operators
+		PullSecret:       swag.String(pullSecret),
+	}
+
+	if cluster.Spec.InstallStrategy != nil && cluster.Spec.InstallStrategy.Agent != nil {
+		if len(spec.InstallStrategy.Agent.Networking.ClusterNetwork) > 0 {
+			clusterParams.ClusterNetworkCidr = swag.String(spec.InstallStrategy.Agent.Networking.ClusterNetwork[0].CIDR)
+			clusterParams.ClusterNetworkHostPrefix = int64(spec.InstallStrategy.Agent.Networking.ClusterNetwork[0].HostPrefix)
+		}
+
+		if len(spec.InstallStrategy.Agent.Networking.ServiceNetwork) > 0 {
+			clusterParams.ServiceNetworkCidr = swag.String(spec.InstallStrategy.Agent.Networking.ServiceNetwork[0])
+		}
```

Comment:
Installer allows for multiple and we support just one?  I guess that's another TODO?

Reply:
Not sure if that a TODO, right now we support only one in rest-api so until we change that we have nothing to do here.
Do you think i should  open tasks to change rest-api, not sure if it's required or we just don't want to break kube-api in the future so it's defined as a list at the moment.